### PR TITLE
Fix 3817 map info url

### PR DESCRIPTION
**Pull request by MV88**
## Description
This pr add the possibility to use actions in queryparams for map viewer

Not all the actions can be dispatched, only the ones in the whiteList [here](https://github.com/geosolutions-it/MapStore2/compare/master...MV88:3817_mapinfo_url?expand=1#diff-36801c0654cb2b242aaa007e85313f75R46)
- SEARCH_LAYER_WITH_FILTER
- ZOOM_TO_EXTENT

whitelist can be configured from localConfig by using the property "initialActionsWhiteList", see [here](https://github.com/geosolutions-it/MapStore2/pull/4000/files?file-filters%5B%5D=.md#diff-6dc5aa664d11eb1277cfda9934324712R31) 

This will be documented in the **API - map query parameters** section

## Issues
 - #3817 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
You cannot dispatch actions from query params for the map viewer

**What is the new behavior?**
You can

**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No (bbox still works)
- 

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Some important point of this pr:
- changes to [identify](https://github.com/geosolutions-it/MapStore2/compare/master...MV88:3817_mapinfo_url?expand=1#diff-e7312b552c083e1cbff765fe521d4d0bR108) epic
- changes to [search](https://github.com/geosolutions-it/MapStore2/compare/master...MV88:3817_mapinfo_url?expand=1#diff-a47d6623b4ded0293bab34bee9d67cf5R220) epic
- moving the share epic into a more generic epic that manages map query params
- changes to [main](https://github.com/geosolutions-it/MapStore2/compare/master...MV88:3817_mapinfo_url?expand=1#diff-059ffaee83c56beb18d35e7692055de4R46) function in order to take the general queryparam epic
this point could be harmful for old projects that are not using main.jsx file

### CODE REVIEW

File: docs/developer-guide/map-query-parameters.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Map query parameters
```
Comment (mbarto) [inline docs/developer-guide/map-query-parameters.md]:
```suggestion
# MapViewer query parameters
```

File: docs/developer-guide/map-query-parameters.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+In this section we will describe the available map query parameters
```
Comment (mbarto) [inline docs/developer-guide/map-query-parameters.md]: I would name these parameters MapViewer parameters, but this is just my opinion.

File: docs/developer-guide/map-query-parameters.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+that can be used when the map is loaded.
+
+## Retro-compatibility
+In the past it has been introduced a **bbox** query param with value that corresponds to extent in the viewport.
```
Comment (mbarto) [inline docs/developer-guide/map-query-parameters.md]:
```suggestion
In the past a **bbox** query param has been introduced whose value corresponds to extent in the viewport.
```

File: docs/developer-guide/map-query-parameters.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+for example:
+`?bbox=-177.84667968750014,-1.8234225930143395,-9.096679687500114,61.700290838326204`
+
+For more details on it see [sharing a map](../user-guide/share).
+
+## New way to dispatch actions in MapStore
```
Comment (mbarto) [inline docs/developer-guide/map-query-parameters.md]:
```suggestion
## Dynamically dispatching initial actions in MapStore
```

File: docs/developer-guide/map-query-parameters.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+**actions** is the new query param and it will be used to dispatch actions (in the map viewer) that are limited to a whitelist.
```
Comment (mbarto) [inline docs/developer-guide/map-query-parameters.md]:
```suggestion
to dispatch additional actions when the map viewer is started, the **actions**  query param can be used. Only actions from a configured whitelist can be dispatched in this way.
```

File: docs/developer-guide/map-query-parameters.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+{
+    "type": "SEARCH:SEARCH_WITH_FILTER",
+    "cql_filter": "ID=75",
+    "layer": "WORKSPACE:LAYER_NAME"
+}
+?actions=[{"type":"SEARCH:SEARCH_WITH_FILTER","cql_filter":"ID=75","layer":"WORKSPACE:LAYER_NAME"}]
+```
+For more details check out the [searchLayerWithFilter](https://mapstore2.geo-solutions.it/mapstore/docs/#actions.search.exports.searchLayerWithFilter) in the framework documentation
+
+#### Add Layers (not implemented yet)
```
Comment (mbarto) [inline docs/developer-guide/map-query-parameters.md]: I would not document something that is not implemented yet. You can add this documentation together with the related implementation.

File: mkdocs.yml [hunk 7]
```diff
@@ -89,4 +89,6 @@ pages:
     - Plugins:
       - Plugins Architecture: 'developer-guide/plugins-architecture.md'
       - Configuring Plugins: 'developer-guide/plugins-documentation.md'
+    - API:
+      - Map query paramaters: 'developer-guide/map-query-parameters.md'
```
Comment (mbarto) [inline mkdocs.yml]:
```suggestion
      - MapViewer query paramaters: 'developer-guide/map-query-parameters.md'
```

File: web/client/actions/__tests__/mapInfo-test.js [hunk 8]
```diff
@@ -89,6 +90,20 @@ describe('Test correctness of the map actions', () => {
         });
     });
 
+    it('reset reverse geocode data', () => {
```
Comment (mbarto) [inline web/client/actions/__tests__/mapInfo-test.js]: Is the test name correct?

File: web/client/actions/map.js [hunk 9]
```diff
@@ -143,9 +152,10 @@ function updateVersion(version) {
     };
 }
 
-function initMap() {
+function initMap(location) {
     return {
-        type: INIT_MAP
+        type: INIT_MAP,
+        location
```
Comment (mbarto) [inline web/client/actions/map.js]: Why this addition?

File: web/client/epics/queryparams.js [hunk 10]
```diff
@@ -37,6 +41,16 @@ const paramActions = {
                 position: "tc"
             })
         ];
+    },
+    actions: ({value = ''}) => {
+        const whiteList = [
```
Comment (mbarto) [inline web/client/epics/queryparams.js]: I would like this whitelist to be configurable from localConfig.json

File: web/client/actions/map.js (hunk 9 above)
Comment (MV88) [reply web/client/actions/map.js]: good catch, it was there from a first implementation and it is no longer used.
going to fix it

File: web/client/actions/__tests__/mapInfo-test.js (hunk 8 above)
Comment (MV88) [reply web/client/actions/__tests__/mapInfo-test.js]: nope XD

File: docs/developer-guide/local-config.md [hunk 11]
```diff
@@ -27,6 +27,8 @@ This is the main structure:
   // API keys for bing and mapquest services
   "bingApiKey",
   "mapquestApiKey",
+  // list of actions that are available to be launched dynamically from query param (#3817)
+  "initialActionsWhiteList": [{"type":"ZOOM_TO_EXTENT", "extent":[minx,minx,maxx,maxy]}],
```
Comment (mbarto) [inline docs/developer-guide/local-config.md]: This is a list of strings (the list of action types, not a list of objects)

**Pull request merged**